src/OpenTelemetry.Extensions.Hosting/OpenTelemetryServicesExtensions.cs

```diff
@@ -50,6 +51,11 @@ public static IServiceCollection AddOpenTelemetryTracing(this IServiceCollection
                 throw new ArgumentNullException(nameof(configure));
             }
 
+            if (services.Any(service => service.ImplementationType == typeof(TelemetryHostedService)))
+            {
+                throw new InvalidOperationException("AddOpenTelemetryTracing must be only called once.");
```

Comment:
I feel like a `NotSupportedException` might be slightly better here, something like:

```csharp
throw new NotSupportedException("Registering multiple TracerProviders via AddOpenTelemetryTracing is not supported.");
```